keras_cv/models/object_detection/faster_rcnn.py

```diff
@@ -0,0 +1,361 @@
+        self.fc_dims = fc_dims
+        self.convs = []
+        for conv_dim in conv_dims:
+            layer = tf.keras.layers.Conv2D(
+                filters=conv_dim,
+                kernel_size=3,
+                strides=1,
+                padding="same",
+                activation="relu",
+            )
+            self.convs.append(layer)
+        self.fcs = []
+        for fc_dim in fc_dims:
+            layer = tf.keras.layers.Dense(units=fc_dim, activation="relu")
+            self.fcs.append(layer)
+        self.box_pred = tf.keras.layers.Dense(units=4)
+        self.cls_score = tf.keras.layers.Dense(units=classes + 1)
+
+    def call(self, feature_map):
+        x = feature_map
+        for conv in self.convs:
+            x = conv(x)
+        for fc in self.fcs:
+            x = fc(x)
+        rcnn_boxes = self.box_pred(x)
+        rcnn_scores = self.cls_score(x)
+        return rcnn_boxes, rcnn_scores
+
+    def get_config(self):
+        config = {
+            "classes": self.num_classes,
+            "conv_dims": self.conv_dims,
+            "fc_dims": self.fc_dims,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+# TODO(tanzheny): add more configurations
+class FasterRCNN(tf.keras.Model):
+    """A Keras model implementing the FasterRCNN architecture.
+
+    Implements the FasterRCNN architecture for object detection.  The constructor
+    requires `classes`, `bounding_box_format` and a `backbone`.
+
+    Usage:
+    ```python
+    retina_net = keras_cv.models.FasterRCNN(
+        classes=20,
+        bounding_box_format="xywh",
+        backbone="resnet50",
+        include_rescaling=False,
+    )
+    ```
+
+    Args:
+        classes: the number of classes in your dataset excluding the background
+            class.  Classes should be represented by integers in the range
+            [0, classes).
+        bounding_box_format: The format of bounding boxes of input dataset. Refer
+            [to the keras.io docs](https://keras.io/api/keras_cv/bounding_box/formats/)
+            for more details on supported bounding box formats.
+        backbone: Either `"resnet50"` or a custom backbone model.
+        include_rescaling: Required if provided backbone is a pre-configured model.
+            If set to `True`, inputs will be passed through a `Rescaling(1/255.0)`
+            layer. Default to False.
+    """
+
+    def __init__(
+        self,
+        classes,
+        bounding_box_format,
+        backbone=None,
+        include_rescaling=False,
+        **kwargs,
+    ):
+        self.bounding_box_format = bounding_box_format
+        super().__init__(**kwargs)
+        self.anchor_generator = AnchorGenerator(
+            bounding_box_format="yxyx",
+            sizes={2: 16.0, 3: 32.0, 4: 64.0, 5: 128.0, 6: 256.0},
+            # scales=[2**x for x in [0, 1 / 3, 2 / 3]],
+            scales=[2**x for x in [0]],
```

Comment:
The scale is misaligned with the comment.

Reply:
Done.